clippy_lints/src/mixed_locale_idents.rs

```diff
@@ -33,56 +20,138 @@ declare_clippy_lint! {
     /// The following code compiles without any warnings:
     ///
     /// ```rust
-    /// struct Blоck { // Not a common latin `o` used, but rather Russian `о`.
-    ///     _щука: String, // Usage of `щ` will suppress `mixed_script_confusables` warning.
-    /// }
+    /// // In the `Block` identifier part, it's not a common latin `o` used, but rather Russian `о`.
+    /// // It's hard to see visually and the `mixed_script_confusables` warning is not spawned
+    /// // because of the `Блок` identifier part.
+    /// struct BlоckБлок;
     ///
     /// fn main() {
-    ///     let _block = Blоck { _щука: "pike".to_string() };
+    ///     let _block = BlоckБлок;
     /// }
     /// ```
     ///
     /// The same example, but with `Block` (english `o`) used instead of `Blоck` (russian `о`).
-    /// It will not compile
+    /// It will not compile.
     ///
     /// ```compile_fail
-    /// struct Blоck {
-    ///     _щука: String,
-    /// }
+    /// struct BlоckБлок;
     ///
     /// fn main() {
-    ///     let _block = Block { _щука: "pike".to_string() };
-    ///
+    ///     let _block = BlockБлок;
     /// }
     ///
     /// // Compile output:
     /// //
-    /// //    error[E0422]: cannot find struct, variant or union type `Block` in this scope
-    /// //    --> src/main.rs:6:18
-    /// //     |
-    /// //   1 | struct Blоck {
-    /// //     | ------------ similarly named struct `Blоck` defined here
-    /// //   ...
-    /// //   6 |     let _block = Block { _щука: "pike".to_string() };
-    /// //     |                  ^^^^^
-    /// //     |
-    /// //   help: a struct with a similar name exists
-    /// //     |
-    /// //   6 |     let _block = Blоck { _щука: "pike".to_string() };
-    /// //     |                  ^^^^^
-    /// //   help: consider importing one of these items
-    /// //
+    /// //  error[E0425]: cannot find value `BlockБлок` in this scope
+    /// //   --> src/main.rs:4:18
+    /// //    |
+    /// //  1 | struct BlоckБлок;
+    /// //    | ----------------- similarly named unit struct `BlоckБлок` defined here
+    /// //  ...
+    /// //  4 |     let _block = BlockБлок;
+    /// //    |                  ^^^^^^^^^ help: a unit struct with a similar name exists: `BlоckБлок`
     /// ```
     pub MIXED_LOCALE_IDENTS,
     style,
-    "multiple locales used in a single identifier"
+    "multiple locales used in a single identifier part"
 }
 
 #[derive(Clone, Debug)]
 pub struct MixedLocaleIdents;
 
 impl_lint_pass!(MixedLocaleIdents => [MIXED_LOCALE_IDENTS]);
 
+/// Cases that normally can be met in the Rust identifiers.
+#[derive(Debug)]
+enum Case {
+    /// E.g. `SomeStruct`, delimiter is uppercase letter.
+    Camel,
```

Comment:
Okay so a problem with this is that not all writing systems have uppercase. More thought needs to be put into how that will work.

Reply:
But perhaps using only mixed script confusables without underscores in a character is enough for us for now. Idk.